.github/workflows/sonarqube.yml

```
@@ -18,10 +18,35 @@
+          python -m pip install --upgrade pip
+          pip install -r backend/requirements.txt
+      
+      - name: Create Firebase Credentials File
+        run: echo "${{ secrets.FIREBASE_CREDENTIALS_JSON }}" > backend/firebase-credentials.json
+        shell: bash # Ensure bash shell for proper echo behavior
+
+      - name: Create Backend .env File
+        env:
+          BACKEND_ENV_CONTENT: ${{ secrets.BACKEND_ENV_FILE }}
+        run: printf "%s\n" "$BACKEND_ENV_CONTENT" > backend/.env
+        shell: bash # Use printf for safer handling of special characters
```

YAMLlint (1.35.1)

[warning] too few spaces before comment

(comments)